qemu/qmp.go

```diff
@@ -1176,7 +1176,10 @@ func (q *QMP) ExecHotplugMemory(ctx context.Context, qomtype, id, mempath string
 		"props":    map[string]interface{}{"size": uint64(size) << 20},
 	}
 	if mempath != "" {
-		args["mem-path"] = mempath
+		props, ok := args["props"].(map[string]interface{})
+		if ok {
```

Comment:
It would be best to avoid the type assertion and the extra if altogether.  We could do this by storing props in a separate variable at the top of the function, e.g.,

```
props := map[string]interface{}{"size": uint64(size) << 20}
```

args would be

```
 	args := map[string]interface{}{
 		"qom-type": qomtype
 		"id":       id,
		"props":    props,
	}		
```

and then

```
if mempath != "" {
    props["mem-path"] = mempath
}
```

Reply:
Oh yes I do prefer this code layout.

Reply:
Good point. I will remember that next time.The commit has been updated.